subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -164,6 +165,46 @@ static int zsock_getaddrinfo_internal(const char *host, const char *service,
 	return 0;
 }
 
+#ifdef CONFIG_USERSPACE
+Z_SYSCALL_HANDLER(z_zsock_getaddrinfo_internal, host, service, hints, res)
+{
+	struct zsock_addrinfo hints_copy;
+	char *host_copy = NULL, *service_copy = NULL;
+	u32_t ret;
+
+	if (hints) {
```

Comment:
This may be minor, but what's the ordering pattern of processing params here? I'd think they should be processed in the original order, or maybe group by in/out-ness, original order in each group. Here, the order appears random. (I pat attention to such things because if I had to do that, I'd copy-pasted existing code, and existing code doesn't try to teach to do it in some consistent pattern.)

Reply:
There's no particular ordering requirement, other than to validate pointers and copy any containing information before using it.